src/Parsing/Impl/Ast/INode.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Python.Core.Text;
+
+namespace Microsoft.Python.Parsing.Ast {
+    public interface INode {
```

Comment:
public interfaces should not have setters, especially on things that do not change like indices which are positions in the buffer AST was produced from.

Reply:
ok!